pilot/docker/Dockerfile.pilot

```diff
@@ -12,6 +12,7 @@ USER 1337:1337
 # The following section is used as base image if BASE_DISTRIBUTION=distroless
 # hadolint ignore=DL3007
 FROM gcr.io/distroless/static:nonroot as distroless
+WORKDIR /
```

Comment:
redundant `WORKDIR /` results in a small ineffeciency.

Reply:
It's not redundant at all: it means this image will actually run.

My previous PR (#20311) changed the Deployments so that the containers run as user/group 1337:1337.
Kubernetes (or Docker on its own) can't successfully run a container from this image when setting user/group to 1337:1337 – see [this comment](#issuecomment-582148470) – because the distroless base image has set `WORKDIR /home/nonroot` and user 1337 has no permissions to reach that directory. By changing the workdir to `/` we fix this problem.

Really, this fix should have gone in #20311 but I didn't discover this problem then due to a difference in behaviour with kind compared to Kubernetes. Kind only showed up the same error once I'd configured the container's security context to drop all capabilities (i.e. this PR). I'm planning to raise an issue on the kind project shortly.

Reply:
I think @sdake's point is that we're already doing this 2 lines below in this particular `Dockerfile`. It's still required in the other `Dockerfile`s, I don't think there's any question about that.

Reply:
Ah, my apologies @sdake – the minimal context shown in the PR view meant I didn't see this! Will remove this line & push.

Reply:
@michaelbannister - thanks. I didn't realize the context wasn't shown or I would have offered more information in the review. Using GitHub is a bit robotic for me 😁